File: 104.Maximum_Depth_Of_A_Binary_Tree/main.go
package main

import "fmt"

type TreeNode struct {
    Val   int
    Left  *TreeNode
    Right *TreeNode
}

type Solution struct{}

func (s Solution) maxDepth(root *TreeNode) int {
    if root == nil {
        return 0
    }
    leftDepth := s.maxDepth(root.Left)
    rightDepth := s.maxDepth(root.Right)
    if leftDepth > rightDepth {
        return leftDepth + 1
    }
    return rightDepth + 1
}

func main() {
    root := &TreeNode{Val: 3}
    root.Left = &TreeNode{Val: 9}
    root.Right = &TreeNode{Val: 20, Left: &TreeNode{Val: 15}, Right: &TreeNode{Val: 7}}

    solution := Solution{}
    fmt.Println("Maximum Depth:", solution.maxDepth(root)) // Output: 3
}

File: 104.Maximum_Depth_Of_A_Binary_Tree/main.ts
class TreeNode {
     val: number;
     left: TreeNode | null;
     right: TreeNode | null;
     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
         this.val = val === undefined ? 0 : val;
         this.left = left === undefined ? null : left;
         this.right = right === undefined ? null : right;
     }
 }
 
 class Solution {
     public maxDepth(root: TreeNode | null): number {
         if (root === null) {
             return 0;
         }
         const leftDepth = this.maxDepth(root.left);
         const rightDepth = this.maxDepth(root.right);
         return Math.max(leftDepth, rightDepth) + 1;
     }
 }
 
 const root = new TreeNode(3, new TreeNode(9), new TreeNode(20, new TreeNode(15), new TreeNode(7)));
 const solution = new Solution();
 console.log("Maximum Depth:", solution.maxDepth(root)); // Output: 3
 